tests/decorator-free.test.ts

```diff
@@ -0,0 +1,419 @@
+
+      const steps = (await DBOS.listWorkflowSteps(handle.workflowID))!;
+      expect(steps.length).toBe(1);
+      expect(steps[0].functionID).toBe(0);
+      expect(steps[0].name).toBe('stepTestStatic');
+      expect(steps[0].output).toEqual(1000);
+      expect(steps[0].error).toBeNull();
+      expect(steps[0].childWorkflowID).toBeNull();
+    } finally {
+      await client.destroy();
+    }
+  });
+
+  test('wf-static-step-run', async () => {
+    const wfid = randomUUID();
+
+    await DBOS.withNextWorkflowID(wfid, async () => {
+      const res = await TestClass.wfRunStepStatic(10);
+      expect(res).toBe(1000);
+    });
+
+    const status = await DBOS.getWorkflowStatus(wfid);
+    expect(status).not.toBeNull();
+    expect(status!.workflowName).toBe('TestClass.wfRunStepStatic');
+
+    const steps = (await DBOS.listWorkflowSteps(wfid))!;
+    expect(steps.length).toBe(1);
+    expect(steps[0].functionID).toBe(0);
+    expect(steps[0].name).toBe('stepTest-runStep');
+    expect(steps[0].output).toEqual(1000);
+    expect(steps[0].error).toBeNull();
+    expect(steps[0].childWorkflowID).toBeNull();
+  });
+
+  test('wf-static-step-retry', async () => {
+    const wfid = randomUUID();
+
+    await DBOS.withNextWorkflowID(wfid, async () => {
+      const res = await TestClass.wfRegRetryStatic(10);
+      expect(res).toBe(1000);
+    });
+
+    const status = await DBOS.getWorkflowStatus(wfid);
+    expect(status).not.toBeNull();
+    expect(status!.workflowName).toBe('TestClass.wfRegRetryStatic');
+
+    const steps = (await DBOS.listWorkflowSteps(wfid))!;
+    expect(steps.length).toBe(1);
+    expect(steps[0].functionID).toBe(0);
+    expect(steps[0].name).toBe('retryTestStatic');
+    expect(steps[0].output).toEqual(1000);
+    expect(steps[0].error).toBeNull();
+    expect(steps[0].childWorkflowID).toBeNull();
+
+    expect(TestClass.retryTestAttempts).toEqual([false, true, true, true, false]);
+  });
+
+  test('wf-inst-step-reg', async () => {
+    const wfid = randomUUID();
+    await DBOS.withNextWorkflowID(wfid, async () => {
+      const res = await inst.wfRegStep(10);
+      expect(res).toBe(1000);
+    });
+
+    const status = await DBOS.getWorkflowStatus(wfid);
+    expect(status).not.toBeNull();
+    expect(status!.workflowName).toBe('TestClass.prototype.wfRegStep');
+
+    const steps = (await DBOS.listWorkflowSteps(wfid))!;
+    expect(steps.length).toBe(1);
+    expect(steps[0].functionID).toBe(0);
+    expect(steps[0].name).toBe('stepTest');
+    expect(steps[0].output).toEqual(1000);
+    expect(steps[0].error).toBeNull();
+    expect(steps[0].childWorkflowID).toBeNull();
+  });
+
+  test('wf-inst-step-reg-queue', async () => {
+    const client = await DBOSClient.create(config.databaseUrl!);
+    try {
+      const handle = await client.enqueue<typeof TestClass.prototype.wfRegStep>(
+        {
+          queueName: queue.name,
+          workflowClassName: 'TestClass',
```

Comment:
Because of how we register configured instance workflows, the caller has to specify the configured class name for registered instance methods (but not for registered static methods). I believe we should investigate relaxing this requirement. It feels redundant to have to specify the class name when you're explicitly specifying the workflow name. However, whatever we do there should be in a separate PR.

Reply:
The workflow name might be ambiguous, no?  I think you need both class and workflow name to uniquely identify.